### PR TITLE
v23.2.0-alpha.4 Release Notes

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-7787

### CODE REVIEW

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+<h3 id="v23-2-0-alpha-4-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- Renamed [cluster settings](../v23.2/cluster-settings.html) related to physical cluster replication for consistency. For example, `bulkio.stream_ingestion.minimum_flush_interval` is now `physical_replication.consumer.minimum_flush_interval`. [#111197][#111197]
+
+<h3 id="v23-2-0-alpha-4-sql-language-changes">SQL language changes</h3>
+
+- [`SHOW SCHEDULES`](../v23.2/show-schedules.html) has two columns that surface the schedule options. These columns have been renamed to align with the documented option names: `on_previous_running` and `on_execution_failure`. [#111759][#111759]
+- Added support for the PLpgSQL `CLOSE` statement, which allows a PLpgSQL routine to close a cursor with the name specified by a cursor variable. [#111330][#111330]
+- When a [`RESTORE`](../v23.2/restore.html) with `remove_regions` is performed, the restore job will now fail if the object contains a [`REGIONAL BY ROW`](../v23.2/multiregion-overview.html#table-locality) table. [#111443][#111443]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:19]: Would it be clearer if the link to "REGIONAL BY ROW" went to this page: https://www.cockroachlabs.com/docs/v23.1/table-localities#regional-by-row-tables?

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+- It is now possible to open a [cursor](../v23.2/cursors.html) within a PLpgSQL function or procedure with an exception block. If an error occurs, creation of the cursor is rolled back before control reaches the exception handler. [#111735][#111735]
+- If a [scheduled backup](../v23.2/create-schedule-for-backup.html) resumes on a new cluster (e.g., after physical cluster replication cutover or a cluster restore), the backup schedule will pause. The user may [resume the schedule](../v23.2/resume-schedules.html) without changing it, but should take special care to ensure no other schedule is backing up to the same [collection](../v23.2/take-full-and-incremental-backups.html#backup-collections). The user may also want to cancel the paused schedule and start a new one. [#111578][#111578]
+- Added support for PLpgSQL `FETCH` and `MOVE` statements. Similar to SQL `FETCH`/`MOVE` statements, commands that would seek the [cursor](../v23.2/cursors.html) backward will fail. In addition, expressions other than constant integers are not yet supported for the `count` option. [#111318][#111318]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:22]: link "FETCH" to https://www.postgresql.org/docs/current/plpgsql-cursors.html#PLPGSQL-CURSOR-USING-FETCH
link "MOVE" to  https://www.postgresql.org/docs/current/plpgsql-cursors.html#PLPGSQL-CURSOR-USING-MOVE

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+- Added support for the `REFCURSOR` data type. `REFCURSOR` is a special string type that is used to handle cursors. PLpgSQL cursor declarations are required to use a variable of type `REFCURSOR`, and the name of a cursor can be passed to and from a PLpgSQL function or procedure. [#111392][#111392]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:23]: link "REFCURSOR data type" to https://www.postgresql.org/docs/current/plpgsql-cursors.html#PLPGSQL-CURSOR-DECLARATIONS

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+- Added two changes to `FOR UPDATE`:
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:24]: link "FOR UPDATE" to https://www.cockroachlabs.com/docs/v23.2/select-for-update

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    For example, the following query is now allowed because the subquery using the prohibited operations is not affected by the `FOR UPDATE OF`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SELECT * FROM t, (SELECT DISTINCT 0, 0 UNION SELECT a, count(*) FROM t GROUP BY a HAVING a > 0) AS u FOR UPDATE OF t;
+    ~~~
+
+    This matches PostgreSQL. [#111258][#111258]
+- Identifiers after numeric constants that are not separated by whitespace are now disallowed to match PostgreSQL 15 behavior. [#112021][#112021]
+- Added the new column `contention_type` to the [`crdb_internal.transaction_contention_events`](../v23.2/crdb-internal.html) table. This column indicates the type of transaction contention encountered. Current values are `LOCK_WAIT` and `SERIALIZATION_CONFLICT`. These contention events specify that contention occurred between two transactions due to serialization conflicts. [#111685][#111685]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:78]: link "transaction contention" to https://www.cockroachlabs.com/docs/v23.1/performance-best-practices-overview#transaction-contention

you may remove the sentence "These contention events specify that contention occurred between two transactions due to serialization conflicts."

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+- Changed the error message: `statement error cannot execute FOR UPDATE in a read-only transaction` to `statement error cannot execute SELECT FOR UPDATE in a read-only transaction` to match PostgreSQL. [#112138][#112138]
+- Added a new [session variable](../v23.2/set-vars.html) `optimizer_use_lock_op_for_serializable`, which when set enables a new implementation of `SELECT FOR UPDATE`. This new implementation of `SELECT FOR UPDATE` acquires row locks **after** any joins and filtering, and always acquires row locks on the primary index of the table being locked. This more closely matches `SELECT FOR UPDATE` behavior in PostgreSQL, but at the cost of more round trips from gateway node to replica leaseholder. Under read-committed isolation (and other isolation levels weaker than serializable), CockroachDB will always use this new implementation of `SELECT FOR UPDATE` regardless of the value of `optimizer_use_lock_op_for_serializable` to ensure correctness. [#112138][#112138]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:80]: link "SELECT FOR UPDATE" to https://www.cockroachlabs.com/docs/v23.2/select-for-update

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+- Fixed a bug where `indoption` inside `pg_index` was not properly encoded causing clients to be unable to decode it as `int2vector`. [#111911][#111911]
+- This patch fixes an issue where the [optimizer](../v23.2/cost-based-optimizer.html) fails to honor the `statement_timeout` session setting when generating constrained index scans for queries with large `IN` lists or `= ANY` predicates on multiple index key columns, which may lead to an out of memory condition on the node. [#111979][#111979]
+- This patch fixes a performance issue in join queries with a `LIMIT` clause, where the [optimizer](../v23.2/cost-based-optimizer.html) may fail to push a `WHERE` clause filter into a join due to how the `LIMIT` operation is internally rewritten. This causes a full scan of the table referenced in the filter. [#110593][#110593]
+- Fixed a bug that caused internal errors during query optimization in rare cases. The bug has been present since version v2.1.11, but it is more likely to occur in version v21.2.0 and later, though it is still rare. The bug only presents when a query contains `min` and `max` aggregate functions. [#112255][#112255]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:145]: link "aggregate functions" to https://www.cockroachlabs.com/docs/v23.2/functions-and-operators#aggregate-functions

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+<h3 id="v23-2-0-alpha-4-performance-improvements">Performance improvements</h3>
+
+- This patch adds support for insert fast-path uniqueness checks on [`REGIONAL BY ROW`](../v23.2/multiregion-overview.html#table-localities) tables where the source is a `VALUES` clause with a single row. This results in a reduction in latency for single-row inserts to `REGIONAL BY ROW` tables and hash-sharded `REGIONAL BY ROW` tables with unique indexes. [#111822][#111822]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md:149]: link "REGIONAL BY ROW" to https://www.cockroachlabs.com/docs/v23.1/table-localities#regional-by-row-tables

**Review by florence-crl — APPROVED** [review]
LGTM
Had some non-blocking suggestions for links to add

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 8 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done!

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 7 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 6 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 5 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 4 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md (hunk 1 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+- Updated the `licenses/CCT.txt` file to reflect the latest [Cockroachdb Community License](https://www.cockroachlabs.com/cockroachdb-community-license/). [#112494][#112494]
+
+<h3 id="v23-2-0-alpha-4-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- Renamed [cluster settings](../v23.2/cluster-settings.html) related to physical cluster replication for consistency. For example, `bulkio.stream_ingestion.minimum_flush_interval` is now `physical_replication.consumer.minimum_flush_interval`. [#111197][#111197]
+
+<h3 id="v23-2-0-alpha-4-sql-language-changes">SQL language changes</h3>
+
+- [`SHOW SCHEDULES`](../v23.2/show-schedules.html) has two columns that surface the schedule options. These columns have been renamed to align with the documented option names: `on_previous_running` and `on_execution_failure`. [#111759][#111759]
+- Added support for the PLpgSQL `CLOSE` statement, which allows a PLpgSQL routine to close a cursor with the name specified by a cursor variable. [#111330][#111330]
```
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md]: done

**Comment by kathancox** [comment]
Thanks for the review @florence-crl, all links incorporated!

**Pull request merged**